src/modules/job-list/list.c

```diff
@@ -180,7 +268,21 @@ void list_cb (flux_t *h, flux_msg_handler_t *mh,
                              "constraint", &constraint) < 0) {
         seterror (&err, "invalid payload: %s", flux_msg_last_error (msg));
         errno = EPROTO;
-        goto error;
+    }
+    if (!constraint) {
+        /* Since "constraint" object is optional, we can't check legacy
+         * RPC format only if EPROTO above.  Double check that legacy RPC
+         * fields weren't supplied instead.
```

Comment:
I stumbled on this comment the first time reading it. It might read better stated more simply like:

> Double check for legacy RPC fields since "constraint" object is optional in new protocol:

Though it could just be me so feel free to leave it as is.